src/ext/java/org/opentripplanner/ext/siri/SiriTripPatternCache.java

```diff
@@ -78,16 +82,17 @@ public synchronized TripPattern getOrCreateTripPattern(
         .getPatternForTrip()
         .get(trip);
 
-      tripPattern.setCreatedByRealtimeUpdater();
+      tripPatternBuilder.withCreatedByRealtimeUpdater(true);
+      tripPatternBuilder.withOriginalTripPattern(originalTripPattern);
 
+      tripPattern = tripPatternBuilder.build();
       // Copy information from the TripPattern this is replacing
       if (originalTripPattern != null) {
-        tripPattern.setOriginalTripPattern(originalTripPattern);
         tripPattern.setHopGeometriesFromPattern(originalTripPattern);
       }
 
       // Add pattern to cache
-      cache.put(key, tripPattern);
+      cache.put(key, tripPatternBuilder.build());
```

Comment:
Shouldn't this be `tripPattern` (with updated geometries)?

Reply:
true, bug introduced when I added back originalTripPattern